src/components/common/FormPage/UpdatePassword.tsx

```diff
@@ -0,0 +1,252 @@
+"use client";
+
+import { zodResolver } from "@hookform/resolvers/zod";
+import { Eye, EyeOff } from "lucide-react";
+import { ChangeEvent, FormEvent, useState } from "react";
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+
+import {
+  AlertDialog,
+  AlertDialogAction,
+  AlertDialogContent,
+  AlertDialogDescription,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogTitle,
+  AlertDialogTrigger,
+} from "~/components/ui/alert-dialog";
+import { Button } from "~/components/ui/button";
+import {
+  Form,
+  FormControl,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "~/components/ui/form";
+import { Input } from "~/components/ui/input";
+import PasswordChecks from "./PasswordChecks";
+
+interface VisibilityState {
+  currentPassword: boolean;
+  newPassword: boolean;
+  confirmPassword: boolean;
+}
+
+const passwordSchema = z
+  .object({
+    password: z
+      .string()
+      .min(8, "Password must be at least 8 characters")
+      .regex(/[A-Z]/, "Password must contain at least one uppercase letter")
+      .regex(/\d/, "Password must contain at least one number")
+      .regex(
+        /[!"#$%&()*,.:<>?@^{|}]/,
+        "Password must contain at least one special character",
+      ),
+    confirmPassword: z.string(),
+  })
+  .refine((data) => data.password === data.confirmPassword, {
+    message: "Passwords do not match",
+    path: ["confirmPassword"],
+  });
+
+const PasswordForm: React.FC = () => {
+  const [password, setPassword] = useState<string>("");
+  const [confirmPassword, setConfirmPassword] = useState<string>("");
```

Comment:
You are using password and confirmPassword only in handleSubmit.
Other cases, you are using the value directly from the form. There is no need to set state, if you are going to be reading directly from the form. Reduce the need to reload your component. See Controlled Vs Uncontrolled Inputs

Reply:
My bad, I was coding half asleep.

Reply:
I need to use controlled input so i can set password checks